activemodel/lib/active_model/type/registry.rb

```diff
@@ -1,70 +1,37 @@
 # frozen_string_literal: true
 
 module ActiveModel
-  # :stopdoc:
   module Type
-    class Registry
+    class Registry # :nodoc:
       def initialize
-        @registrations = []
+        @registrations = {}
       end
 
-      def initialize_dup(other)
+      def initialize_copy(other)
         @registrations = @registrations.dup
         super
       end
 
-      def register(type_name, klass = nil, **options, &block)
+      def register(type_name, klass = nil, &block)
         unless block_given?
           block = proc { |_, *args| klass.new(*args) }
```

Comment:
I wonder if we could just do `block = klass.method(:new)`.

Reply:
We have to drop the first param which is the type name. Not sure why it's there tbh…

Reply:
oh right, totally missed it...